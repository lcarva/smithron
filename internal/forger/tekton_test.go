package forger

import (
	"context"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestTekton(t *testing.T) {
	p := Plan{
		Name:    "hello-world",
		Image:   "docker.io/busybox:latest",
		Command: []string{"echo", "Hello, World!"},
		Env: map[string]string{
			"GREETING": "Hello",
		},
	}

	ctx := context.Background()

	tf, err := GetForgerForTarget("tekton")
	require.NoError(t, err)
	require.IsType(t, TektonForger{}, tf)

	got, err := tf.Forge(ctx, p)
	require.NoError(t, err)
	snaps.MatchYAML(t, got)
}
