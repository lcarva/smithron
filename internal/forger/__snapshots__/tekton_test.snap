
[TestTekton - 1]
apiVersion: tekton.dev/v1
kind: Task
metadata:
  creationTimestamp: null
  name: hello-world
spec:
  steps:
  - command:
    - echo
    - Hello, World!
    computeResources: {}
    env:
    - name: GREETING
      value: Hello
    image: docker.io/busybox:latest
    name: run

---
